internal/cmd/ssh.go

```diff
@@ -167,72 +171,125 @@ func setupDestinationSSHConfig(ctx context.Context, cli *CLI, destination *api.D
 		return err
 	}
 
-	if err := provisionSSHKey(ctx, cli, client, infraSSHDir); err != nil {
+	user, err := client.GetUserSelf(ctx)
+	if err != nil {
+		return err
+	}
+
+	keyFilename, err := provisionSSHKey(ctx, provisionSSHKeyOptions{
+		cli:         cli,
+		client:      client,
+		hostConfig:  hostCfg,
+		infraSSHDir: infraSSHDir,
+		user:        user,
+	})
+	if err != nil {
 		return fmt.Errorf("create ssh keypair: %w", err)
 	}
 
 	if err := writeInfraKnownHosts(infraSSHDir, destination); err != nil {
 		return fmt.Errorf("write known hosts: %w", err)
 	}
 
-	user, err := client.GetUserSelf(ctx)
-	if err != nil {
-		return err
-	}
-
-	if err := writeDestinationSSHConfig(infraSSHDir, destination, user); err != nil {
+	if err := writeDestinationSSHConfig(infraSSHDir, destination, user, keyFilename); err != nil {
 		return fmt.Errorf("write infra ssh config: %w", err)
 	}
 	return nil
 }
 
-func provisionSSHKey(ctx context.Context, cli *CLI, client *api.Client, infraSSHDir string) error {
-	keyFilename := filepath.Join(infraSSHDir, "key")
+type provisionSSHKeyOptions struct {
+	cli         *CLI
+	client      *api.Client
+	infraSSHDir string
+	hostConfig  *ClientHostConfig
+	user        *api.User
+}
+
+func provisionSSHKey(ctx context.Context, opts provisionSSHKeyOptions) (string, error) {
+	keysCfg, err := readKeysConfig(opts.infraSSHDir)
+	switch {
+	case errors.Is(err, fs.ErrNotExist):
+		keysCfg = &keysConfig{}
+	case err != nil:
+		return "", err
+	}
 
-	// TODO: check expiration
-	// TODO: check the key exists in the API
-	if fileExists(keyFilename) {
-		return nil
+	org, err := opts.client.GetOrganizationSelf(ctx)
```

Comment:
It would be nice to get this from the `hostConfig` instead of having to query for it again. If the `LoginResponse` included this ID in addition to the name, I think it would be easy to store this in the `hostConfig`. 

I created #3827 for this.